src/datajudge/constraints/stats.py

```diff
@@ -0,0 +1,47 @@
+from typing import Any, Optional, Tuple
+
+import sqlalchemy as sa
+from scipy.stats import ks_2samp
+
+from .. import db_access
+from ..db_access import DataReference
+from .base import Constraint, OptionalSelections
+
+
+class KolmogorovSmirnov2Sample(Constraint):
+    def __init__(
+        self, ref: DataReference, ref2: DataReference, significance_level: float = 0.05
+    ):
+        self.significance_level = significance_level
+        super().__init__(ref, ref2=ref2)
+
+    @staticmethod
+    def calculate_2sample_ks_test(data, data2):
```

Comment:
```suggestion
    def _calculate_2sample_ks_test(data, data2) -> float:
```

You might want to try to annotate `data` and `data2`, but don't think it's critical.

Reply:
I have added numpy.typing.ArrayLike, though I'm not sure if that's the best approach as I haven't seen it being used in many other places. Makes sense, though, since that is exactly what the scipy method expects (when viewing the docs)

Reply:
I agree that it's in principle a good approach. My only concern is what would happen when someone wants to execute any other constraint in `datajudge` without having `numpy` installed. Will the declaration of the types that don't exist blow up at runtime?

My guess is no, given the `if` guard I suggested for the import. But I have no 100% certainty. What I'm saying is that perhaps you can execute some constraints while removing `numpy` from the environment.

Reply:
I agree with you on all points but I don't think it's worth the effort. We are perfectly fine using Python's `Iterable` or `Collection` type hints.